src/errors/FileNotReadableError.ts

```diff
@@ -0,0 +1,6 @@
+export class FileNotReadableError extends Error {
+  constructor(filename?: string) {
+    super('File ' + filename + ' exists, but cannot be read.');
```

Comment:
You can use string template instead of concatenation. 

```suggestion
    super(`File ${filename} exists, but cannot be read.`);
```